ocaml/xapi/test_extauth_plugin_ADpbis.ml

```diff
@@ -55,9 +55,39 @@ module PbisAuthErrorsCatch= Generic.Make(struct
     ]
   end)
 
+module PbisExtractSid = Generic.Make(struct
+    module Io = struct
+      type input_t = (string * string) list
+      type output_t = string list
+
+      let string_of_input_t = Test_printers.(list (pair string string))
+      let string_of_output_t = Test_printers.(list string)
+    end
+
+    let transform = Extauth_plugin_ADpbis.extract_sid_from_group_list
+
+    let tests = [
+      [(" ", " ")], [];
+
+      [("Number of groups found for user 'admin@NVC'", "1");
+       ("", "Group[1 of 1] name = NVC\\testg(ab) (gid = 564135020, sid = S-1-5-21-1171552557-368733809-2946345504-1132)")],
+      ["S-1-5-21-1171552557-368733809-2946345504-1132"];
+
+      [("Number of groups found for user 'cnk3@UN'", "1");
+       ("", "Group[1 of 1] name = UN\\KnmOJ (gid = 492513842, sid = S-1-5-31-5921451325-154521381-3135732118-4527)")],
+      ["S-1-5-31-5921451325-154521381-3135732118-4527"];
+
+      [("Number of groups found for user 'test@testdomain'", "2");
+       ("", "Group[1 of 2] name = testdomain\\dnsadmins (gid = 580912206, sid = S-1-5-21-791009147-1041474540-2433379237-1102)");
+       ("", "Group[2 of 2] name = testdomain\\domain+users (gid = 580911617, sid = S-1-5-21-791009147-1041474540-2433379237-513)")],
+      ["S-1-5-21-791009147-1041474540-2433379237-1102"; "S-1-5-21-791009147-1041474540-2433379237-513"];
+    ]
+  end)
```

Comment:
Is it supposed to get '1' really? I would imagine that that should be considered broken input...